src/command_line_parser.h

```diff
@@ -0,0 +1,317 @@
+  int64_t pinned_memory_pool_byte_size_{1 << 28};
+  std::list<std::pair<int, uint64_t>> cuda_pools_;
+
+  // [FIXME] this option is broken after backend separation: this should have
+  // controlled backend copy behavior but not properly propagate to backend
+  // after separation, need to go through backend config.
+  int32_t buffer_manager_thread_count_{0};
+
+  std::vector<std::tuple<std::string, std::string, std::string>> host_policies_;
+
+  // Cache configuration
+  bool enable_cache_{false};
+  std::string cache_dir_{"/opt/tritonserver/caches"};
+  std::unordered_map<
+      std::string, std::vector<std::pair<std::string, std::string>>>
+      cache_config_settings_;
+
+#ifdef TRITON_ENABLE_LOGGING
+  bool log_info_{true};
+  bool log_warn_{true};
+  bool log_error_{true};
+  int32_t log_verbose_{0};
+  triton::common::Logger::Format log_format_{
+      triton::common::Logger::Format::kDEFAULT};
+  std::string log_file_{};
+#endif  // TRITON_ENABLE_LOGGING
+
+#ifdef TRITON_ENABLE_TRACING
+  std::string trace_filepath_{};
+  TRITONSERVER_InferenceTraceLevel trace_level_{
+      TRITONSERVER_TRACE_LEVEL_DISABLED};
+  int32_t trace_rate_{1000};
+  int32_t trace_count_{-1};
+  int32_t trace_log_frequency_{0};
+#endif  // TRITON_ENABLE_TRACING
+
+// The configurations for various endpoints (i.e. HTTP, GRPC and metrics)
+#ifdef TRITON_ENABLE_HTTP
+  bool allow_http_{true};
+  std::string http_address_{"0.0.0.0"};
+  int32_t http_port_{8000};
+  bool reuse_http_port_{false};
+  // The number of threads to initialize for the HTTP front-end.
+  int http_thread_cnt_{8};
+#endif  // TRITON_ENABLE_HTTP
+
+#ifdef TRITON_ENABLE_GRPC
+  bool allow_grpc_{true};
+  triton::server::grpc::Options grpc_options_;
+#endif  // TRITON_ENABLE_GRPC
+
+#ifdef TRITON_ENABLE_METRICS
+  bool allow_metrics_{true};
+  // Note that socket address is not part of metrics config,
+  // current implementation enforce metric to use the same address as in
+  // HTTP endpoint.
+  int32_t metrics_port_{8002};
+  // Metric settings for Triton core
+  float metrics_interval_ms_{2000};
+  bool allow_gpu_metrics_{true};
+  bool allow_cpu_metrics_{true};
+#endif  // TRITON_ENABLE_METRICS
+
+#ifdef TRITON_ENABLE_SAGEMAKER
+  bool allow_sagemaker_{false};
+  std::string sagemaker_address_{"0.0.0.0"};
+  int32_t sagemaker_port_{8080};
+  bool sagemaker_safe_range_set_{false};
+  std::pair<int32_t, int32_t> sagemaker_safe_range_{-1, -1};
+  // The number of threads to initialize for the SageMaker HTTP front-end.
+  int sagemaker_thread_cnt_{8};
+#endif  // TRITON_ENABLE_SAGEMAKER
+
+#ifdef TRITON_ENABLE_VERTEX_AI
+  bool allow_vertex_ai_{false};
+  std::string vertex_ai_address_{"0.0.0.0"};
+  int32_t vertex_ai_port_{8080};
+  // The number of threads to initialize for the Vertex AI HTTP front-end.
+  int vertex_ai_thread_cnt_{8};
+  std::string vertex_ai_default_model_{};
+#endif  // TRITON_ENABLE_VERTEX_AI
+
+  // [FIXME] who should call this function?
+  void CheckPortCollision();
```

Comment:
Is this comment still valid?

Reply:
This will probably be a helper function in `main.cc`, once the parser is broken down, the collision checking will be done after all endpoint parsing is done so can't be part of any option struct